base/field_abilities/PoundFieldPsynergy.ts

```diff
@@ -0,0 +1,266 @@
+            this.hand_tween(this.hand_translate_x, this.hand_translate_y, 350, () => {
+                this.game.add.tween(this.hand_sprite).to({width: 0, height: 0}, 350, Phaser.Easing.Linear.None, true);
+                this.hand_translate_x = this.controllable_char.sprite.centerX;
+                this.hand_translate_y = this.controllable_char.sprite.centerY;
+                this.hand_tween(this.hand_translate_x, this.hand_translate_y, 400, () => {
+                    this.unset_hero_cast_anim();
+                    this.stop_casting();
+                    this.data.overlayer_group.remove(this.hand_sprite, false);
+                });
+            });
+        });
+        timer.start();
+    }
+
+    //Change hand movement depending on the cast direction
+    move_hand_by_dir(final_x, final_y) {
+        switch (this.cast_direction) {
+            case directions.up:
+                final_y -= 12;
+                final_x *= 0;
+                break;
+            case directions.down:
+                final_y += 4;
+                final_x *= 0;
+                break;
+            case directions.left:
+                final_x *= -1;
+                break;
+        }
+        this.hand_translate_x += final_x;
+        this.hand_translate_y += final_y;
+    }
+
+    hand_tween(var_x, var_y, time, complete_action) {
+        this.game.add
+            .tween(this.hand_sprite)
+            .to({centerX: var_x, centerY: var_y}, time, Phaser.Easing.Linear.None, true)
+            .onComplete.addOnce(complete_action);
+    }
+
+    init_stars() {
+        for (let i = 0; i < PoundFieldPsynergy.STAR_MAX_COUNT; ++i) {
+            const initial_x = this.hand_sprite.centerX + Math.cos((Math.PI + numbers.degree60) * i);
+            const initial_y = this.hand_sprite.centerY + Math.sin((Math.PI + numbers.degree60) * i);
+            let star_sprite = this.data.overlayer_group.create(initial_x, initial_y, "star");
+            const this_angle = ((Math.PI + numbers.degree90) * i) / (12 - 1) + numbers.degree30;
+            const final_x = initial_x + Math.cos(this_angle) * 48;
+            const final_y = initial_y + Math.sin(this_angle) * 36;
+            const tween = this.game.add
+                .tween(star_sprite)
+                .to({x: final_x, y: final_y}, 250, Phaser.Easing.Linear.None, true, i * (Phaser.Timer.QUARTER / 25));
+            tween.onComplete.addOnce(() => {
+                star_sprite.destroy();
+            });
+        }
+    }
+
+    init_wave(x_pos, y_pos) {
+        const size = 64;
+        let wave = new Wave(this.game, x_pos, y_pos, this.wave_width, this.wave_height);
+        wave.phase = 0;
+        wave.transparent_radius = 0;
+        this.data.npc_group.add(wave.sprite);
+        wave.sprite.sendToBack();
+        wave.update();
+
+        let wave_animation = this.game.add
+            .tween(this)
+            .to({wave_width: size, wave_height: size}, 300, Phaser.Easing.Linear.None);
```

Comment:
Instead of controlling `wave_width` and `wave_height`, control `wave.sprite.width` and `wave.sprite.height`, so you don't need to set `wave_animation.onUpdateCallback`.

Reply:
This doesn't work, It still need to calle both the resize and update functions of wave on update or It wont resize

Reply:
I created a `size` property for the `Wave` class, then you don't need to create `wave_width` and `wave_height`, please merge with master. Also, when you change the `size` property, the wave updates automatically. Also in this commit, I already set the sprite anchor to 0.5 on wave construction. With these modifications check if you also can reduce the number of tweens.